experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/master/AmActor.scala

```diff
@@ -258,36 +255,37 @@ class AmActor(appConfig: AppConfig, yarnConf: YarnConfiguration) extends Actor {
     done.reason match {
       case failed: Failed =>
         val message = s"Failed. total=${appConfig.getEnv(WORKER_CONTAINERS).toInt}, completed=${stats.completed}, allocated=${stats.allocated}, failed=${stats.failed}"
-        amRMClient ! AMStatusMessage(FinalApplicationStatus.FAILED, message, null)
+        rmClientActor ! AMStatusMessage(FinalApplicationStatus.FAILED, message, null)
         success = false
       case ShutdownRequest =>
         if (stats.failed == 0 && stats.completed == appConfig.getEnv(WORKER_CONTAINERS).toInt) {
           val message = s"ShutdownRequest. total=${appConfig.getEnv(WORKER_CONTAINERS).toInt}, completed=${stats.completed}, allocated=${stats.allocated}, failed=${stats.failed}"
-          amRMClient ! AMStatusMessage(FinalApplicationStatus.KILLED, message, null)
+          rmClientActor ! AMStatusMessage(FinalApplicationStatus.KILLED, message, null)
           success = false
         } else {
           val message = s"ShutdownRequest. total=${appConfig.getEnv(WORKER_CONTAINERS).toInt}, completed=${stats.completed}, allocated=${stats.allocated}, failed=${stats.failed}"
-          amRMClient ! AMStatusMessage(FinalApplicationStatus.FAILED, message, null)
+          rmClientActor ! AMStatusMessage(FinalApplicationStatus.FAILED, message, null)
           success = false
         }
        case AllRequestedContainersCompleted =>
         val message = s"Diagnostics. total=${appConfig.getEnv(WORKER_CONTAINERS).toInt}, completed=${stats.completed}, allocated=${stats.allocated}, failed=${stats.failed}"
-        amRMClient ! AMStatusMessage(FinalApplicationStatus.SUCCEEDED, message, null)
+        rmClientActor ! AMStatusMessage(FinalApplicationStatus.SUCCEEDED, message, null)
         success = true
     }
 
-    amRMClient ! PoisonPill
+    rmClientActor ! PoisonPill
     success
     }
+
 }
- 
-class RMCallbackHandlerActor(appConfig: AppConfig, yarnAM: ActorRef) extends Actor {
+
+class RMCallbackHandlerActor(appConfig: AppConfig) extends Actor {
```

Comment:
The function in this class is very simple, only sending a message with no event handler defined.

Is it truely necessary to define it as an actor?